src/providers/wfs/oapif/qgsoapifitemsrequest.cpp

```diff
@@ -150,74 +163,160 @@ void QgsOapifItemsRequest::processReply()
+        {
+          mNumberMatched = numberMatched.get<int>();
+        }
       }
     }
+    catch ( const json::parse_error &ex )
+    {
+      mErrorCode = QgsBaseNetworkRequest::ApplicationLevelError;
+      mAppLevelError = ApplicationLevelError::JsonError;
+      mErrorMessage = errorMessageWithReason( tr( "Cannot decode JSON document: %1" ).arg( QString::fromStdString( ex.what() ) ) );
+      emit gotResponse();
+      return;
+    }
   }
-  catch ( const json::parse_error &ex )
+  else
   {
-    mErrorCode = QgsBaseNetworkRequest::ApplicationLevelError;
-    mAppLevelError = ApplicationLevelError::JsonError;
-    mErrorMessage = errorMessageWithReason( tr( "Cannot decode JSON document: %1" ).arg( QString::fromStdString( ex.what() ) ) );
-    emit gotResponse();
-    return;
+    // Get numberMatched and next link from HTTP response headers
+    for ( const auto &headerKeyValue : mResponseHeaders )
+    {
+      if ( headerKeyValue.first.compare( QByteArray( "OGC-NumberMatched" ), Qt::CaseSensitivity::CaseInsensitive ) == 0 )
+      {
+        bool ok = false;
+        const int val = QString::fromUtf8( headerKeyValue.second ).toInt( &ok );
+        if ( ok )
+          mNumberMatched = val;
+      }
+      else if ( headerKeyValue.first.compare( QByteArray( "Link" ), Qt::CaseSensitivity::CaseInsensitive ) == 0 )
+      {
+        // Parse stuff like:
+        //  <https://ogc-api.nrw.de/lika/v1/collections/flurstueck/items?f=html>; rel="alternate"; title="This document as HTML"; type="text/html", <https://ogc-api.nrw.de/lika/v1/collections/flurstueck/items?f=fgb&offset=10>; rel="next"; title="Next page"; type="application/flatgeobuf"
+
+        // Split on commas, except when they are in double quotes or between <...>, and skip padding space before/after separator
+        const thread_local QRegularExpression splitOnComma( R"(\s*,\s*(?=(?:[^"<]|"[^"]*"|<[^>]*>)*$))" );
+        const QStringList links = QString::fromUtf8( headerKeyValue.second ).split( splitOnComma );
+        for ( const QString &link : links )
+        {
+          if ( link.isEmpty() || link[0] != QLatin1Char( '<' ) )
+            continue;
+          const int idxClosingBracket = static_cast<int>( link.indexOf( QLatin1Char( '>' ) ) );
+          if ( idxClosingBracket < 0 )
+            continue;
+          const QString href = link.mid( 1, idxClosingBracket - 1 );
+          const int idxSemiColon = static_cast<int>( link.indexOf( QLatin1Char( ';' ), idxClosingBracket ) );
+          if ( idxSemiColon < 0 )
+            continue;
+          // Split on semi-colon, except when they are in double quotes, and skip padding space before/after separator
+          const thread_local QRegularExpression splitOnSemiColon( R"(\s*;\s*(?=(?:[^"]*"[^"]*")*[^"]*$))" );
+          const QStringList linkParts = link.mid( idxSemiColon + 1 ).split( splitOnSemiColon );
+          QString rel, type;
+          for ( const QString &linkPart : linkParts )
```

Comment:
```suggestion
          for ( const QString &linkPart : std::as_const( linkParts ) )
```

Redudndant, I know but I was instructed to do so by top QGIS core developers and I agree that it makes the intent clearer.

Reply:
fixed